### DISCUSSION TITLE
Extend the Gutenberg UI to wp-admin

### DISCUSSION THREAD

**Discussion by tomxygen** [WordPress/gutenberg#47531] [Ideas]
WordPress now features two very different design languages. The new modern and minimal Gutenberg design language and the old wp-admin style.
Expanding the Gutenberg design language to wp-admin would be great to unify the overall experience.
Are there any plans to do so?

**Comment by robglidden** [comment]
I believe Yoast founder argued recently that the WordPress admin sorely needs a design overhaul.

[Yoast SEO Founder: WordPress Admin Interface Is “Simply Bad”](https://www.searchenginejournal.com/wordpress-admin-ui-is-bad/477612/#close)

On Twitter,[ it was proposed that Gutenberg has a component library](https://twitter.com/jdevalk/status/1618152402489856003) that could be used, but Yoast quickly pointed out "that’s not done. Not even close."

I for one would love a new admin overhaul, but I would prefer something more versatile, useful, modern, forward-looking and integratabtle than a circa-2017 bag of React components.

I don't think every WordPress plugin should have reengineer to React/Redux just to improve the admin.  I vote web components or a generalizable design system, as some more advanced/modern design systems are doing.

**Comment by HikeMike** [comment]
I've been looking at building an admin theme based on the site editor. Most of it can be achieved in css over the existing markup. For instance, the toolbar can be given a white background and it's contents made about twice the height. Inputs, buttons and such could extend the scss classes from Gutenberg. There are certainly places components can be used as well though. I like the Navigation component for the sidebar, and I know the WooCommerce team have built something similar for their screens. You could perhaps json_encode the $menu and $submenu globals here, so Filters still work. I certainly think it's worth exploring.

**Reply by tomxygen** [reply]
That would be awesome!
I’d love to help you build that and test if you’re actually going to build it.

**Comment by HikeMike** [comment]
Apologies for the delay. I've started playing with a repo but it's still a few weeks off being shareable 🙃 I'm now thinking it makes more sense to add @wordpress/interface (ie Gutenberg) to the admin pages. We can then replace the Toolbar with a Header which has scoped slots, and replace the Sidebar with a Component based one, similar to the Site Editor. This also provides a consistent Data Store and potential support for creating component based admin screens, in a similar way to how you can add a sidebar now. I'll keep you posted.

**Reply by tomxygen** [reply]
Thank you!
I'd love to see screenshots of the implementation

**Comment by HikeMike** [comment]
It seems this is now more formally in the pipeline. https://make.wordpress.org/core/2023/07/12/admin-design/
